system-addon/lib/TopSitesFeed.jsm

```diff
@@ -18,26 +17,24 @@ this.TopSitesFeed = class TopSitesFeed {
   constructor() {
     this.lastUpdated = 0;
   }
-  getData(action) {
-    return Task.spawn(function*() {
-      let newAction;
-      let links = yield PlacesProvider.links.getLinks();
-      let result;
-      if (!links) {
-        result = [];
-      } else {
-        result = links.filter(link => link && link.type !== "affiliate").slice(0, 12);
-      }
-      newAction = {type: at.TOP_SITES_UPDATED, data: result};
-      // send an update to content so the preloaded tab can get the udpated content
-      this.store.dispatch(ac.SendToContent(newAction, action.meta.fromTarget));
-      this.lastUpdated = Date.now();
-      for (let link of links) {
-        let screenshot = yield PreviewProvider.getThumbnail(link.url);
-        newAction = {type: at.SCREENSHOT_UPDATED, data: {url: link.url, screenshot}};
-        this.store.dispatch(ac.BroadcastToContent(newAction));
-      }
-    }.bind(this));
+  async getData(action) {
+    let newAction;
+    let links = await PlacesProvider.links.getLinks();
+    let result;
+    if (!links) {
+      result = [];
+    } else {
+      result = links.filter(link => link && link.type !== "affiliate").slice(0, 12);
+    }
+    newAction = {type: at.TOP_SITES_UPDATED, data: result};
+    // send an update to content so the preloaded tab can get the udpated content
+    this.store.dispatch(ac.SendToContent(newAction, action.meta.fromTarget));
+    this.lastUpdated = Date.now();
+    for (let link of links) {
+      let screenshot = await PreviewProvider.getThumbnail(link.url);
```

Comment:
i think for now we want to do this sequentially for now because it's the only thing we can show for top sites right now (since we don't have high res icons, tippy top, or a fallback)